### PR TITLE
mexico nuevo leon

### LINKED ISSUE #443: Mexico Nuevo Leon spider
**Links to bulk download and/or API documentation**
http://si.nl.gob.mx/transparencia/acerca-del-proyecto

**Link to CRM issue**
https://crm.open-contracting.org/issues/5662

**Priority**
anytime

### PR DESCRIPTION
**Pull request by yolile**
closes #443

### CODE REVIEW

**Comment by yolile** [comment]
@jpmckinney assigning this to you as there is a change in the ZipSpider base class that I want to check with you before updating the tests.

File: kingfisher_scrapy/base_spider.py [hunk 1]
```diff
@@ -304,31 +308,40 @@ def start_requests(self):
...
+        else:
+            compressed_file = rarfile.RarFile(BytesIO(response.body))
+        for finfo in compressed_file.infolist():
+            filename = util.get_base_filename(finfo.filename)
+            if self.file_name_contains not in filename:
+                continue
+            if self.compression == 'rar':
+                file_info = compressed_file.getinfo(finfo)
+                if file_info.isdir():
+                    continue
```
Comment (jpmckinney) [inline kingfisher_scrapy/base_spider.py:329]: We can do this for ZIP files, too. However, the method name is `is_dir` instead (annoyingly).

File: kingfisher_scrapy/base_spider.py [hunk 2]
```diff
@@ -304,31 +308,40 @@ def start_requests(self):
 
     encoding = 'utf-8'
     zip_file_format = None
+    compression = 'zip'
+    file_name_contains = ''
```
Comment (jpmckinney) [inline kingfisher_scrapy/base_spider.py:312]: Rename this to `file_name_must_contain` to make the semantics clearer.

**Comment by yolile** [comment]
all good, thank you!

**Pull request merged**